Source/DOM classes/SVG-DOM/SVGHelperUtilities.m

```diff
@@ -423,76 +425,120 @@ +(CALayer *) newCALayerForPathBasedSVGElement:(SVGElement<SVGTransformable>*) sv
 		if( actualLineCap.length > 0 )
 		{
 			if( [actualLineCap isEqualToString:@"butt"] )
-				_shapeLayer.lineCap = kCALineCapButt;
+				strokeLayer.lineCap = kCALineCapButt;
 			else if( [actualLineCap isEqualToString:@"round"] )
-				_shapeLayer.lineCap = kCALineCapRound;
+				strokeLayer.lineCap = kCALineCapRound;
 			else if( [actualLineCap isEqualToString:@"square"] )
-				_shapeLayer.lineCap = kCALineCapSquare;
+				strokeLayer.lineCap = kCALineCapSquare;
 		}
 		if( actualLineJoin.length > 0 )
 		{
 			if( [actualLineJoin isEqualToString:@"miter"] )
-				_shapeLayer.lineJoin = kCALineJoinMiter;
+				strokeLayer.lineJoin = kCALineJoinMiter;
 			else if( [actualLineJoin isEqualToString:@"round"] )
-				_shapeLayer.lineJoin = kCALineJoinRound;
+				strokeLayer.lineJoin = kCALineJoinRound;
 			else if( [actualLineJoin isEqualToString:@"bevel"] )
-				_shapeLayer.lineJoin = kCALineJoinBevel;
+				strokeLayer.lineJoin = kCALineJoinBevel;
 		}
 		if( actualMiterLimit.length > 0 )
 		{
-			_shapeLayer.miterLimit = [actualMiterLimit floatValue];
+			strokeLayer.miterLimit = [actualMiterLimit floatValue];
 		}
+		if ( [actualStroke hasPrefix:@"url"] )
+		{
+			// need a new fill layer because the stroke layer is becoming a mask
```

Comment:
Please duplicate / reference this comment on line 372. Otherwise, it looks like a bug / typo that you have two references to the same object. I can see what you're doing, seems good, but would help a lot to make this clear.